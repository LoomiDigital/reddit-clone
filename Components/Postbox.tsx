import React, { useState } from "react";
import { useSession } from "next-auth/react";
import { useForm } from "react-hook-form";
import { toast } from "react-hot-toast";
import { LinkIcon, PhotoIcon } from "@heroicons/react/24/outline";
import {
  PostAttributesFragmentDoc,
  useAddPostMutation,
  useAddSubredditMutation,
  useAddVoteMutation,
  useGetSubredditByTopicLazyQuery,
} from "@d20/generated/graphql";

import Avatar from "./Avatar";

type Props = {
  subreddit?: string;
};

type FormData = {
  postTitle: string;
  postBody: string;
  postImage: string;
  subreddit: string;
};

function Postbox({ subreddit }: Props) {
  const { data: session } = useSession();
  const [addPost] = useAddPostMutation();
  const [addVote] = useAddVoteMutation();
  const [addSubreddit] = useAddSubredditMutation();
  const [getSubReddit] = useGetSubredditByTopicLazyQuery();

  const {
    register,
    handleSubmit,
    setValue,
    watch,
    formState: { errors },
  } = useForm<FormData>();
  const [imageBoxOpen, setImageBoxOpen] = useState<boolean>(false);

  const submitPost = handleSubmit(async (formData) => {
    const notification = toast.loading("Creating post...");

    try {
      const { data } = await getSubReddit({
        variables: {
          topic: subreddit || formData.subreddit,
        },
        fetchPolicy: "no-cache",
      });

      const subredditExists = data?.getSubredditByTopic;

      const postFields = {
        title: formData.postTitle!,
        body: formData.postBody || "",
        image: formData.postImage || "",
        username: session?.user.name!,
      };

      let newSubreddit = null;

      if (!subredditExists) {
        const { data: addSubredditData } = await addSubreddit({
          variables: {
            topic: formData.subreddit.toLowerCase(),
          },
        });
        newSubreddit = addSubredditData?.insertSubreddit!;
      }

      addPost({
        variables: {
          ...postFields,
          subreddit_id: newSubreddit?.id || subredditExists?.id!,
          subreddit_topic: newSubreddit?.topic || subredditExists?.topic!,
        },
        optimisticResponse(vars) {
          return {
            insertPost: {
              __typename: "Post",
              id: -1,
              title: vars.title,
              body: vars.body,
              image: vars.image,
              username: vars.username,
              votes: [],
              subreddit_id: -1,
              subreddit_topic: vars.subreddit_topic,
              created_at: new Date().toISOString(),
            },
          };
        },
        update: async (cache, { data: addPostData }) => {
          const fieldName = subreddit ? "postsByTopic" : "posts";

          cache.modify({
            fields: {
              [fieldName](existingPosts = {}) {
                const newPostRef = cache.writeFragment({
                  data: {
                    ...addPostData?.insertPost!,
                    votes: [
                      {
                        __typename: "Vote",
                        id: 666,
                        username: session?.user.name!,
                        upvote: true,
                      },
                    ],
                  },
                  fragment: PostAttributesFragmentDoc,
                  fragmentName: "postAttributes",
                });

                return {
                  ...existingPosts,
                  edges: [
                    {
                      __typename: "PostEdge",
                      node: newPostRef,
                      cursor: addPostData?.insertPost?.id!,
                    },
                    ...existingPosts.edges,
                  ],
                };
              },
            },
          });
        },
        onCompleted: (data) => {
          addVote({
            variables: {
              post_id: data.insertPost?.id!,
              username: session?.user.name!,
              upvote: true,
            },
          });
        },
      });

      setValue("postTitle", "");
      setValue("postBody", "");
      setValue("postImage", "");
      setValue("subreddit", "");

      toast.success("Post created!", { id: notification });
    } catch (error) {
      console.log("error", error);
      toast.error("Error creating post!", { id: notification });
    }
  });

  return (
    <form
      className="sticky top-16 z-50 rounded-md border border-gray-300 bg-white p-2"
      onSubmit={submitPost}
    >
      <div className="flex items-center space-x-3">
        <Avatar />

        <input
          {...register("postTitle", { required: true })}
          disabled={!session}
          className="w-full flex-1 rounded-md bg-gray-50 p-2 pl-5 outline-none"
          type="text"
          placeholder={
            session
              ? subreddit
                ? `Create a post in r/${subreddit}`
                : "Create a post by entering a title!"
              : "Sign in to create a post"
          }
        />
        <PhotoIcon
          onClick={() => {
            setImageBoxOpen(!imageBoxOpen);
          }}
          className={`h-6 cursor-pointer text-gray-300 ${
            imageBoxOpen && "text-blue-500"
          }`}
        />
        <LinkIcon className="h-6 text-gray-300" />
      </div>
      {!!watch("postTitle") && (
        <div className="flex flex-col py-2">
          <div className="flex items-center px-2">
            <p className="min-w-[90px]">Body:</p>
            <input
              className="m-2 flex-1 bg-blue-50 p-2 outline-none"
              {...register("postBody")}
              type="text"
              placeholder="Text (optional)"
            />
          </div>
          {!subreddit && (
            <div className="flex flex-col py-2">
              <div className="flex items-center px-2">
                <p className="min-w-[90px]">Subreddit:</p>
                <input
                  className="m-2 flex-1 bg-blue-50 p-2 outline-none"
                  {...register("subreddit", { required: true })}
                  type="text"
                  placeholder="i.e. r/nextjs"
                />
              </div>
            </div>
          )}
          {imageBoxOpen && (
            <div className="flex flex-col py-2">
              <div className="flex items-center px-2">
                <p className="min-w-[90px]">Image URL:</p>
                <input
                  className="m-2 flex-1 bg-blue-50 p-2 outline-none"
                  {...register("postImage")}
                  type="text"
                  placeholder="Optional..."
                />
              </div>
            </div>
          )}
          {Object.keys(errors).length > 0 && (
            <div className="space-y-2 p-2 text-red-500">
              {errors.postTitle?.type === "required" && (
                <p>- Post title is required</p>
              )}
              {errors.subreddit?.type === "required" && (
                <p>- A subreddit is required</p>
              )}
            </div>
          )}
          {!!watch("postTitle") && (
            <button
              type="submit"
              className="w-full rounded-full bg-blue-500 p-2 text-white"
            >
              Create Post
            </button>
          )}
        </div>
      )}
    </form>
  );
}

export default Postbox;
