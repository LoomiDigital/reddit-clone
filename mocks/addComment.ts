import { MockedResponse } from "@apollo/client/testing";
import { AddCommentDocument } from "@d20/generated/graphql";

export const mockAddComment: MockedResponse = {
  request: {
    query: AddCommentDocument,
    variables: {
      post_id: 1,
      text: "This is a new comment",
      username: "Buck",
    },
  },
  result: {
    data: {
      addComment: {
        id: 3,
        text: "This is a new comment",
        post_id: 1,
        created_at: "2023-05-08T18:10:10.831966Z",
        username: "Buck",
      },
    },
  },
};
