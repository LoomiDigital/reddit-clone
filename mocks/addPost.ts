import { MockedResponse } from "@apollo/client/testing";
import { AddPostDocument } from "@d20/generated/graphql";

export const mockAddPost: MockedResponse = {
  request: {
    query: AddPostDocument,
    variables: {
      title: "This is a new post",
      body: "A brand new post!",
      image: "https://via.placeholder.com/150",
      username: "Buck",
      subreddit_id: 1,
      subreddit_topic: "testsubreddit",
    },
  },
  result: {
    data: {
      insertPost: {
        __typename: "Post",
        id: 1,
        title: "This is a new post",
        body: "A brand new post!",
        image: "https://via.placeholder.com/150",
        username: "Buck",
        subreddit_id: 1,
        subreddit_topic: "testsubreddit",
        created_at: "2021-08-15T20:00:00.000Z",
        votes: [
          {
            __typename: "Vote",
            id: -1,
            post_id: 1,
            username: "Buck",
            upvote: true,
          },
        ],
      },
    },
  },
};
