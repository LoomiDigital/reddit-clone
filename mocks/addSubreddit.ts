import { MockedResponse } from "@apollo/client/testing";
import { AddSubredditDocument } from "@d20/generated/graphql";

export const mockAddSubreddit: MockedResponse = {
  request: {
    query: AddSubredditDocument,
    variables: {
      topic: "testsubreddit",
    },
  },
  result: {
    data: {
      insertSubreddit: {
        __typename: "Subreddit",
        id: 1,
        post_id: 1,
        topic: "testsubreddit",
        created_at: "2021-08-15T20:00:00.000Z",
      },
    },
  },
};
