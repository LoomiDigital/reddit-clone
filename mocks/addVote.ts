import { MockedResponse } from "@apollo/client/testing";
import { AddVoteDocument } from "@d20/generated/graphql";

export const mockAddVote: MockedResponse = {
  request: {
    query: AddVoteDocument,
    variables: {
      post_id: 1,
      username: "Buck",
      upvote: true,
    },
  },
  result: {
    data: {
      addVote: {
        __typename: "Vote",
        post_id: 1,
        upvote: true,
        username: "Buck",
      },
    },
  },
};
