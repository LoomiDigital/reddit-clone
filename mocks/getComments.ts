import { MockedResponse } from "@apollo/client/testing";
import {
  GetCommentsByPostIdDocument,
  GetPostsDocument,
  GetPostsQuery,
  PostConnection,
} from "@d20/generated/graphql";

export const mockCommentsResponse: MockedResponse = {
  request: {
    query: GetCommentsByPostIdDocument,
    variables: {
      post_id: 1,
    },
  },
  result: {
    data: {
      commentsByPostId: [
        {
          id: 1,
          text: "a comment",
          post_id: 1,
          created_at: "2023-05-08T18:10:10.831966Z",
          username: "barabis",
        },
        {
          id: 2,
          text: "another comment",
          post_id: 1,
          created_at: "2023-05-08T18:10:10.831966Z",
          username: "barabis",
        },
      ],
    },
  },
};

export const mockPosts: GetPostsQuery = {
  posts: {
    __typename: "PostConnection",
    edges: [
      {
        __typename: "PostEdge",
        cursor: "YXJyYXljb25uZWN0aW9uOjA=",
        node: {
          __typename: "Post",
          id: 1,
          title: "Test Post",
          body: "Test Content",
          image: "http://placehold.it/300x300",
          username: "Buck",
          votes: [
            {
              upvote: true,
              username: "Buck",
              post_id: 1,
            },
          ],
          subreddit_id: 1,
          subreddit_topic: "test",
          created_at: "2021-08-15T20:00:00.000Z",
        },
      },
      {
        __typename: "PostEdge",
        cursor: "YXJyYXljb25uZWN0aW9uOjA=",
        node: {
          __typename: "Post",
          id: 2,
          title: "Test Post",
          body: "Test Content",
          username: "Buck",
          image: "http://placehold.it/300x300",
          votes: [
            {
              upvote: true,
              username: "Buck",
              post_id: 2,
            },
          ],
          subreddit_id: 2,
          subreddit_topic: "test",
          created_at: "2021-08-15T20:00:00.000Z",
        },
      },
    ],
    pageInfo: {
      __typename: "PageInfo",
      hasNextPage: false,
      endCursor: "YXJyYXljb25uZWN0aW9uOjA=",
    },
  },
};

export const mockPostConnection: PostConnection = {
  __typename: "PostConnection",
  edges: [
    {
      __typename: "PostEdge",
      cursor: "YXJyYXljb25uZWN0aW9uOjA=",
      node: {
        __typename: "Post",
        id: 1,
        title: "Test Post",
        body: "Test Content",
        image: "http://placehold.it/300x300",
        username: "Buck",
        votes: [
          {
            upvote: true,
            username: "Buck",
            post_id: 1,
          },
        ],
        subreddit_id: 1,
        subreddit_topic: "test",
        created_at: "2021-08-15T20:00:00.000Z",
      },
    },
    {
      __typename: "PostEdge",
      cursor: "YXJyYXljb25uZWN0aW9uOjA=",
      node: {
        __typename: "Post",
        id: 2,
        title: "Test Post",
        body: "Test Content",
        username: "Buck",
        image: "http://placehold.it/300x300",
        votes: [
          {
            upvote: true,
            username: "Buck",
            post_id: 2,
          },
        ],
        subreddit_id: 2,
        subreddit_topic: "test",
        created_at: "2021-08-15T20:00:00.000Z",
      },
    },
  ],
  pageInfo: {
    __typename: "PageInfo",
    hasPreviousPage: false,
    hasNextPage: false,
    startCursor: "YXJyYXljb25uZWN0aW9uOjA=",
    endCursor: "YXJyYXljb25uZWN0aW9uOjA=",
  },
};
