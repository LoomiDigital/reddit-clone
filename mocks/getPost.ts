import { MockedResponse } from "@apollo/client/testing";
import {
  GetPostDocument,
  PostAttributesFragment,
} from "@d20/generated/graphql";

export const mockPostResponse: MockedResponse = {
  request: {
    query: GetPostDocument,
    variables: {
      id: 1,
    },
  },
  result: {
    data: {
      getPost: {
        node: {
          __typename: "Post",
          id: 1,
          title: "Test Post",
          body: "Test Content",
          image: "http://placehold.it/300x300",
          username: "Buck",
          votes: [
            {
              __typename: "Vote",
              upvote: true,
              username: "Buck",
              post_id: 1,
            },
          ],
          subreddit_id: 1,
          subreddit_topic: "test",
          created_at: "2021-08-15T20:00:00.000Z",
        },
      },
    },
  },
};

export const mockPost: PostAttributesFragment = {
  __typename: "Post",
  id: 1,
  title: "Test Post",
  body: "Test Content",
  image: "http://placehold.it/300x300",
  username: "Buck",
  votes: [
    {
      __typename: "Vote",
      upvote: false,
      username: "Buck",
      post_id: 1,
    },
  ],
  subreddit_id: 1,
  subreddit_topic: "test",
  created_at: "2021-08-15T20:00:00.000Z",
};

export const mockPostUpvoted: PostAttributesFragment = {
  __typename: "Post",
  id: 1,
  title: "Test Post",
  body: "Test Content",
  image: "http://placehold.it/300x300",
  username: "Buck",
  votes: [
    {
      __typename: "Vote",
      upvote: true,
      username: "Buck",
      post_id: 1,
    },
  ],
  subreddit_id: 1,
  subreddit_topic: "test",
  created_at: "2021-08-15T20:00:00.000Z",
};
