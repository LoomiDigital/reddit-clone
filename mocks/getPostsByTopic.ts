import { MockedResponse } from "@apollo/client/testing";

import {
  GetPostsByTopicDocument,
  GetPostsByTopicQuery,
} from "@d20/generated/graphql";

export const mockPostsByTopicResponse: MockedResponse = {
  request: {
    query: GetPostsByTopicDocument,
    variables: {
      first: 10,
    },
  },
  result: {
    data: {
      postsByTopic: {
        __typename: "PostConnection",
        edges: [
          {
            __typename: "PostEdge",
            cursor: "YXJyYXljb25uZWN0aW9uOjA=",
            node: {
              __typename: "Post",
              id: 1,
              title: "Test Post",
              body: "Test Content",
              image: "http://placehold.it/300x300",
              username: "Buck",
              votes: [
                {
                  __typename: "Vote",
                  upvote: true,
                  username: "Buck",
                  post_id: 1,
                },
              ],
              subreddit_id: 1,
              subreddit_topic: "test",
              created_at: "2021-08-15T20:00:00.000Z",
            },
          },
          {
            __typename: "PostEdge",
            cursor: "YXJyYXljb25uZWN0aW9uOjA=",
            node: {
              __typename: "Post",
              id: 2,
              title: "Test Post",
              body: "Test Content",
              username: "Buck",
              image: "http://placehold.it/300x300",
              votes: [
                {
                  __typename: "Vote",
                  upvote: true,
                  username: "Buck",
                  post_id: 2,
                },
              ],
              subreddit_id: 2,
              subreddit_topic: "test",
              created_at: "2021-08-15T20:00:00.000Z",
            },
          },
        ],
        pageInfo: {
          __typename: "PageInfo",
          hasNextPage: false,
          endCursor: "YXJyYXljb25uZWN0aW9uOjA=",
        },
      },
    },
  },
};

export const mockPostsByTopic: GetPostsByTopicQuery = {
  postsByTopic: {
    __typename: "PostConnection",
    edges: [
      {
        __typename: "PostEdge",
        cursor: "YXJyYXljb25uZWN0aW9uOjA=",
        node: {
          __typename: "Post",
          id: 1,
          title: "Test Post",
          body: "Test Content",
          image: "http://placehold.it/300x300",
          username: "Buck",
          votes: [
            {
              __typename: "Vote",
              upvote: true,
              username: "Buck",
              post_id: 1,
            },
          ],
          subreddit_id: 1,
          subreddit_topic: "test",
          created_at: "2021-08-15T20:00:00.000Z",
        },
      },
      {
        __typename: "PostEdge",
        cursor: "YXJyYXljb25uZWN0aW9uOjA=",
        node: {
          __typename: "Post",
          id: 2,
          title: "Test Post",
          body: "Test Content",
          username: "Buck",
          image: "http://placehold.it/300x300",
          votes: [
            {
              __typename: "Vote",
              upvote: true,
              username: "Buck",
              post_id: 2,
            },
          ],
          subreddit_id: 2,
          subreddit_topic: "test",
          created_at: "2021-08-15T20:00:00.000Z",
        },
      },
    ],
    pageInfo: {
      __typename: "PageInfo",
      hasNextPage: false,
      endCursor: "YXJyYXljb25uZWN0aW9uOjA=",
    },
  },
};

export const mockUseGetLazyPostsByTopicReturn = {
  posts: mockPostsByTopic.postsByTopic?.edges,
  loading: false,
  hasNextPage: true,
  sentryRef: jest.fn(),
};
