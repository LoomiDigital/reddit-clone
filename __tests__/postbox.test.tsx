import { MockedProvider } from "@apollo/client/testing";
import Postbox from "@d20/components/Postbox";
import { mockAddPost } from "@d20/mocks/addPost";
import { mockAddSubreddit } from "@d20/mocks/addSubreddit";
import { mockAddVote } from "@d20/mocks/addVote";
import {
  mockGetSubredditResponse,
  mockNoSubredditResponse,
  mockSubreddit,
} from "@d20/mocks/getSubreddit";
import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event/";
import { SessionProvider } from "next-auth/react";
import { act } from "react-dom/test-utils";
import { toast } from "react-hot-toast";

describe("Postbox Component", () => {
  const toastSuccess = jest.spyOn(toast, "success");

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should create a new post with existing subreddit from form data", async () => {
    render(
      <MockedProvider
        mocks={[
          mockGetSubredditResponse,
          mockAddSubreddit,
          mockAddPost,
          mockAddVote,
        ]}
      >
        <SessionProvider
          session={{
            expires: "2021-10-10",
            user: {
              name: "Buck",
              expires: "2021-10-10",
              email: "user@test.com",
              address: "123 Fake St",
              image: "https://via.pla ceholder.com/150",
            },
          }}
        >
          <Postbox />
        </SessionProvider>
      </MockedProvider>
    );

    const postTitleInput = screen.getByPlaceholderText(
      "Create a post by entering a title!"
    );
    const imageOpenIcon = screen.getByTitle("Add an image");

    await act(() => {
      fireEvent.click(imageOpenIcon);
    });

    await userEvent.type(postTitleInput, "This is a new post");

    const postSubredditInput = screen.getByPlaceholderText("i.e. r/nextjs");
    const postBodyInput = screen.getByPlaceholderText("Text (optional)");
    const imageInput = screen.getByPlaceholderText("Image URL");

    await userEvent.type(postSubredditInput, "testsubreddit");
    await userEvent.type(imageInput, "https://via.placeholder.com/150");
    await userEvent.type(postBodyInput, "A brand new post!");

    const postButton = await screen.getByRole("button", {
      name: "Create Post",
    });

    await act(() => {
      fireEvent.click(postButton);
    });

    await waitFor(() => {
      expect(toastSuccess).toHaveBeenCalledWith("Post created!", {
        id: "1",
      });
    });
  });

  it("should create a new post with existing subreddit passed in as prop", async () => {
    render(
      <MockedProvider
        mocks={[
          mockGetSubredditResponse,
          mockAddSubreddit,
          mockAddPost,
          mockAddVote,
        ]}
      >
        <SessionProvider
          session={{
            expires: "2021-10-10",
            user: {
              name: "Buck",
              expires: "2021-10-10",
              email: "user@test.com",
              address: "123 Fake St",
              image: "https://via.pla ceholder.com/150",
            },
          }}
        >
          <Postbox subreddit={mockSubreddit.topic} />
        </SessionProvider>
      </MockedProvider>
    );

    const postTitleInput = screen.getByPlaceholderText(
      `Create a post in r/${mockSubreddit.topic}`
    );
    const imageOpenIcon = screen.getByTitle("Add an image");

    await act(() => {
      fireEvent.click(imageOpenIcon);
    });

    await userEvent.type(postTitleInput, "This is a new post");

    const postBodyInput = screen.getByPlaceholderText("Text (optional)");
    const imageInput = screen.getByPlaceholderText("Image URL");

    await userEvent.type(imageInput, "https://via.placeholder.com/150");
    await userEvent.type(postBodyInput, "A brand new post!");

    const postButton = await screen.getByRole("button", {
      name: "Create Post",
    });

    await act(() => {
      fireEvent.click(postButton);
    });

    await waitFor(() => {
      expect(toastSuccess).toHaveBeenCalledWith("Post created!", {
        id: "2",
      });
    });
  });

  it("should create a new post with no existing subreddit", async () => {
    render(
      <MockedProvider
        mocks={[
          mockNoSubredditResponse,
          mockAddSubreddit,
          mockAddPost,
          mockAddVote,
        ]}
      >
        <SessionProvider
          session={{
            expires: "2021-10-10",
            user: {
              name: "Buck",
              expires: "2021-10-10",
              email: "user@test.com",
              address: "123 Fake St",
              image: "https://via.pla ceholder.com/150",
            },
          }}
        >
          <Postbox />
        </SessionProvider>
      </MockedProvider>
    );

    const postTitleInput = screen.getByPlaceholderText(
      "Create a post by entering a title!"
    );
    const imageOpenIcon = screen.getByTitle("Add an image");

    await act(() => {
      fireEvent.click(imageOpenIcon);
    });

    await userEvent.type(postTitleInput, "This is a new post");

    const postSubredditInput = screen.getByPlaceholderText("i.e. r/nextjs");
    const postBodyInput = screen.getByPlaceholderText("Text (optional)");
    const imageInput = screen.getByPlaceholderText("Image URL");

    await userEvent.type(postSubredditInput, "testsubreddit");
    await userEvent.type(imageInput, "https://via.placeholder.com/150");
    await userEvent.type(postBodyInput, "A brand new post!");

    const postButton = await screen.getByRole("button", {
      name: "Create Post",
    });

    await act(() => {
      fireEvent.click(postButton);
    });

    await waitFor(() => {
      expect(toastSuccess).toHaveBeenCalledWith("Post created!", {
        id: "3",
      });
    });
  });
});
