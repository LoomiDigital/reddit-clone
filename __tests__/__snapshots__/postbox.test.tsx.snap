// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Postbox Component should render correctly 1`] = `
<form
  className="sticky top-16 z-50 rounded-md border border-gray-300 bg-white p-2"
  onSubmit={[Function]}
>
  <div
    className="flex items-center space-x-3"
  >
    <div
      className="relative h-10 w-10 overflow-hidden rounded-full border-gray-300 bg-white undefined"
    >
      <img
        alt="Buck"
        data-nimg="fill"
        decoding="async"
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="https://avatars.dicebear.com/api/open-peeps/Buck.svg"
        style={
          {
            "bottom": 0,
            "color": "transparent",
            "height": "100%",
            "left": 0,
            "objectFit": undefined,
            "objectPosition": undefined,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": "100%",
          }
        }
      />
    </div>
    <input
      className="w-full flex-1 rounded-md bg-gray-50 p-2 pl-5 outline-none"
      disabled={false}
      name="postTitle"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Create a post in r/testsubreddit"
      type="text"
    />
    <svg
      aria-hidden="true"
      className="h-6 cursor-pointer text-gray-300 false"
      fill="none"
      onClick={[Function]}
      stroke="currentColor"
      strokeWidth={1.5}
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Add an image
      </title>
      <path
        d="M2.25 15.75l5.159-5.159a2.25 2.25 0 013.182 0l5.159 5.159m-1.5-1.5l1.409-1.409a2.25 2.25 0 013.182 0l2.909 2.909m-18 3.75h16.5a1.5 1.5 0 001.5-1.5V6a1.5 1.5 0 00-1.5-1.5H3.75A1.5 1.5 0 002.25 6v12a1.5 1.5 0 001.5 1.5zm10.5-11.25h.008v.008h-.008V8.25zm.375 0a.375.375 0 11-.75 0 .375.375 0 01.75 0z"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
    <svg
      aria-hidden="true"
      className="h-6 text-gray-300"
      fill="none"
      stroke="currentColor"
      strokeWidth={1.5}
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M13.19 8.688a4.5 4.5 0 011.242 7.244l-4.5 4.5a4.5 4.5 0 01-6.364-6.364l1.757-1.757m13.35-.622l1.757-1.757a4.5 4.5 0 00-6.364-6.364l-4.5 4.5a4.5 4.5 0 001.242 7.244"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  </div>
</form>
`;
